File: api/src/main/java/com/hirrao/javaee/service/BlogService.java
package com.hirrao.javaee.service;

import com.hirrao.javaee.entity.Blog;
import com.baomidou.mybatisplus.core.metadata.IPage;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface BlogService {
    Blog findByBlogId(Long bolgId);

    List<Blog> findByUid(Long uid);

    List<Blog> findByTitle(String title);

    void addBlog(Long blogId, String content, String createTime, String updateTime, Long uid, String title);

    void update(Long blogId, String content, String updateTime, String title);

    void delete(Long blogId);

    IPage<Blog> search(int curPage,int size, Long uid);
}

File: api/src/main/kotlin/cn/hirrao/javaee/service/BlogService.kt
package cn.hirrao.javaee.service

import cn.hirrao.javaee.entity.Blog
import com.baomidou.mybatisplus.core.metadata.IPage
import org.springframework.stereotype.Service

@Service
interface BlogService {
    fun findByBlogId(bolgId: Long?): Blog?

    fun findByUid(uid: Int): List<Blog?>?

    fun findByTitle(title: String?): List<Blog?>?

    fun addBlog(blogId: Long?, content: String?, createTime: String?, updateTime: String?, uid: Int, title: String?)

    fun update(blogId: Long?, content: String?, updateTime: String?, title: String?)

    fun delete(blogId: Long?)

    fun search(curPage: Int, size: Int, uid: Int): IPage<Blog?>?
}
